From 479a32f6c3a068ac8d65669609ad94f967f01d64 Mon Sep 17 00:00:00 2001
From: xiaoleGun <1592501605@qq.com>
Date: Thu, 10 Feb 2022 09:11:53 +0800
Subject: [PATCH] Discard some permission definitions

Although this will break the integrity of the ring test, there is no better way.
---
 sepolicies/28/Honor7x_plat_pub.cil       | 7 -------
 sepolicies/29/Realme_X2_Pro_plat_pub.cil | 1 -
 2 files changed, 8 deletions(-)

diff --git a/sepolicies/28/Honor7x_plat_pub.cil b/sepolicies/28/Honor7x_plat_pub.cil
index 532281e..d9207c3 100644
--- a/sepolicies/28/Honor7x_plat_pub.cil
+++ b/sepolicies/28/Honor7x_plat_pub.cil
@@ -3652,7 +3652,6 @@
 (allow base_typeattr_48_28_0 pdx_performance_client_channel_socket_type (unix_stream_socket (read write getattr setattr lock append getopt setopt shutdown)))
 (allow base_typeattr_48_28_0 pdx_performance_client_server_type (fd (use)))
 (allow pdx_performance_client_server_type base_typeattr_48_28_0 (fd (use)))
-(allow base_typeattr_48_28_0 pdx_bufferhub_client_channel_socket_type (unix_stream_socket (read write getattr setattr lock append getopt setopt shutdown)))
 (allow base_typeattr_48_28_0 pdx_bufferhub_client_server_type (fd (use)))
 (allow pdx_bufferhub_client_server_type base_typeattr_48_28_0 (fd (use)))
 (allow appdomain runas_exec_28_0 (file (getattr)))
@@ -3662,9 +3661,7 @@
 (allow appdomain adbd_28_0 (unix_stream_socket (ioctl read write getattr getopt shutdown)))
 (allow appdomain cache_file_28_0 (dir (getattr)))
 (neverallow base_typeattr_50_28_0 self (capability (chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
-(neverallow base_typeattr_50_28_0 self (capability2 (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
 (neverallow base_typeattr_50_28_0 self (cap_userns (chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
-(neverallow base_typeattr_50_28_0 self (cap2_userns (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
 (neverallow appdomain dev_type (blk_file (read write)))
 (neverallow appdomain audio_device_28_0 (chr_file (read write)))
 (neverallow appdomain camera_device_28_0 (chr_file (read write)))
@@ -5484,9 +5481,7 @@
 (neverallow hal_configstore_server exfat_28_0 (file (ioctl read write create getattr setattr lock relabelfrom relabelto append map unlink link rename execute quotaon mounton execute_no_trans entrypoint execmod open audit_access)))
 (neverallow hal_configstore_server base_typeattr_59_28_0 (service_manager (add find list)))
 (neverallow hal_configstore_server self (capability (chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
-(neverallow hal_configstore_server self (capability2 (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
 (neverallow hal_configstore_server self (cap_userns (chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
-(neverallow hal_configstore_server self (cap2_userns (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
 (neverallow hal_configstore_server base_typeattr_59_28_0 (process (ptrace)))
 (neverallow hal_configstore_server base_typeattr_59_28_0 (file (relabelfrom relabelto)))
 (neverallow hal_configstore_server base_typeattr_59_28_0 (dir (relabelfrom relabelto)))
@@ -7420,8 +7415,6 @@
 (allow runas_28_0 seapp_contexts_file_28_0 (file (ioctl read getattr lock map open)))
 (neverallow runas_28_0 self (capability (chown dac_override dac_read_search fowner fsetid kill setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
 (neverallow runas_28_0 self (cap_userns (chown dac_override dac_read_search fowner fsetid kill setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap)))
-(neverallow runas_28_0 self (capability2 (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
-(neverallow runas_28_0 self (cap2_userns (mac_override mac_admin syslog wake_alarm block_suspend audit_read perf_event)))
 (allow sdcardd_28_0 cgroup_28_0 (dir (ioctl read write create getattr setattr lock rename add_name remove_name reparent search rmdir open)))
 (allow sdcardd_28_0 fuse_device_28_0 (chr_file (ioctl read write getattr lock append map open)))
 (allow sdcardd_28_0 rootfs_28_0 (dir (mounton)))
diff --git a/sepolicies/29/Realme_X2_Pro_plat_pub.cil b/sepolicies/29/Realme_X2_Pro_plat_pub.cil
index 13bc32a..dff7d27 100644
--- a/sepolicies/29/Realme_X2_Pro_plat_pub.cil
+++ b/sepolicies/29/Realme_X2_Pro_plat_pub.cil
@@ -4731,7 +4731,6 @@
 (allow dnsmasq_29_0 netd_29_0 (unix_dgram_socket (read write)))
 (allow dnsmasq_29_0 netd_29_0 (udp_socket (read write)))
 (allow domain init_29_0 (process (sigchld)))
-(allow domain self (process (fork sigchld sigkill sigstop sigcont signull signal getsched setsched getsession getpgid setpgid getcap setcap getattr setrlimit)))
 (allow domain self (fd (use)))
 (allow domain proc_29_0 (dir (ioctl read getattr lock search open)))
 (allow domain proc_net_type (dir (search)))
-- 
2.32.0

